jwe.go

```diff
@@ -187,8 +229,26 @@ func (parsed *rawJSONWebEncryption) sanitized() (*JSONWebEncryption, error) {
 
 	for _, recipient := range obj.recipients {
 		headers := obj.mergedHeaders(&recipient)
-		if headers.getAlgorithm() == "" || headers.getEncryption() == "" {
-			return nil, fmt.Errorf("go-jose/go-jose: message is missing alg/enc headers")
+		alg := headers.getAlgorithm()
+		enc := headers.getEncryption()
+		if !containsKeyAlgorithm(alg, keyAlgorithms) {
+			return nil, fmt.Errorf("go-jose/go-jose: unexpected key algorithm %q; expected %q", alg, keyAlgorithms)
+		}
+		if !containsContentEncryption(enc, contentEncryption) {
+			return nil, fmt.Errorf("go-jose/go-jose: unexpected content encryption algorithm %q; expected %q", enc, contentEncryption)
+		}
```

Comment:
I believe there's a footgun here: if someone passes `[]KeyAlgorithm{""}` or `[]ContentEncryption{""}` to this method, _and_ one of the recipients has no algorithm or no encryption header, these checks will pass. I think it's a good idea to still explicitly error out on objects which have empty headers, so that poorly-crafted arguments to this function can't cause it to accidentally accept them.

The same below.

Reply:
What do you think of explicitly rejecting `[]KeyAlgorithm{""}` (or, specifically, any value in KeyAlgorithm == "")?

Reply:
I think that would be good; warning the person using this library that they've misconfigured something seems like a better feedback loop than telling them that they received a malformed JWE.

But it does also feel a bit like spooky action at a distance: "this loop is safe even if the input is the empty string because ‧.°‧elsewhere‧°.‧ we've ensured that the empty string isn't also in our haystack". It also means that the error message will say `unexpected key algorithm ""; expected...` which is a bit harder to interpret than "message is missing alg header".

Reply:
Good point! Done.